hathor/merged_mining/digibyte_rpc.py

```diff
@@ -0,0 +1,151 @@
+from abc import ABC, abstractmethod
+from itertools import count
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union, cast
+
+from aiohttp import BasicAuth, ClientSession
+from structlog import get_logger
+
+logger = get_logger()
+
+
+class RPCFailure(Exception):
+    def __init__(self, message: str, code: Optional[int] = None):
+        super().__init__(message)
+        self.code = code
+
+
+class IDigibyteRPC(ABC):
```

Comment:
About this class, is it possible to create a parent class with this? For what I see the only difference between this one and the `IBitcoinRPC` are the default capabilities on the `get_block_template`. I might be wrong, but I'm thinking about the other coins that we might add in the future, this could make it easier.

Reply:
I think it makes sense to unify jsonrpc related abstractions, and I'll do that on an `rpc_commons` or even `jsonrpc` module. But I prefer not making a class for common methods, precisely because the little differences are very important, and I also can't foresee what new parent chains (like Bitcoin Cash or SV) will need to be different, so it's safer to assume they all have distinct method sets and implement them independently. But unifying jsonrpc is definitely a plus.

Reply:
Done.